modules/couchbase/module.go

```diff
@@ -0,0 +1,63 @@
+package couchbase
+
+import (
+	"github.com/Trendyol/chaki/as"
+	"github.com/Trendyol/chaki/config"
+	"github.com/Trendyol/chaki/module"
+	"github.com/couchbase/gocb/v2"
+)
+
+var (
+	asClusterOptionsWrapper = as.Struct[ClusterOptionsWrapper]("cbclusteroptionswrapper")
+	asCouchbaseTracers      = as.Interface[gocb.RequestTracer]("gocbrequesttracers")
+)
+
+func Module() *module.Module {
+	m := module.New()
+
+	m.Provide(
+		asClusterOptionsWrapper.Grouper(),
+		asCouchbaseTracers.Grouper(),
+		newCluster,
+	)
+
+	m.AddProvideHook(
+		module.ProvideHook{
+			Match: asClusterOptionsWrapper.Match,
+			Wrap:  asClusterOptionsWrapper.Value,
+		},
+	)
+
+	return m
+}
+
+func newCluster(cfg *config.Config, wrappers []ClusterOptionsWrapper, tracers []gocb.RequestTracer) (*gocb.Cluster, error) {
```

Comment:
Instead of passing wrappers, tracers etc... directly, what if we agree on usage
`func new(cfg *config.Config, ...options)` and differ options from module to module?

Reply:
Make sense. I'll update this module for now later on we can update other modules too